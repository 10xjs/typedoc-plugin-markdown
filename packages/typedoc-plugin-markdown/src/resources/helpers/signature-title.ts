import { SignatureReflection } from 'typedoc';

import { type } from './type';

export function signatureTitle(this: SignatureReflection) {
  const md = [];

  if (this.parent && this.parent.flags) {
    md.push(this.parent.flags.map((flag) => `\`${flag}\``).join(' '));
  }

  if (this.name === '__get') {
    md.push(`**get ${this.parent.name}**`);
  } else if (this.name === '__set') {
    md.push(`**set ${this.parent.name}**`);
  } else if (this.name !== '__call') {
    md.push(`**${this.name}**`);
  }
  if (this.typeParameters) {
    md.push(`‹${this.typeParameters.map((typeParameter) => `**${typeParameter.name}**`).join(', ')}›`);
  }
  const params = this.parameters
    ? this.parameters
        .map((param) => {
          const paramsmd = [];
          if (param.flags.isRest) {
            paramsmd.push('...');
          }
          paramsmd.push(`\`${param.name}`);
          if (param.flags.isOptional) {
            paramsmd.push('?');
          }
          paramsmd.push(`\`: ${type.call(param.type, false)}`);
          return paramsmd.join('');
        })
        .join(', ')
    : '';
  md.push(`(${params})`);
  if (this.type) {
    md.push(`: *${type.call(this.type, false)}*`);
  }
  return md.join('') + '\n';
}
