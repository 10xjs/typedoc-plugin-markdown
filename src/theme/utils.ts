import * as fs from 'fs';
import * as Handlebars from 'handlebars';
import * as path from 'path';
import { DeclarationReflection, Reflection } from 'typedoc/dist/lib/models/reflections/index';

export function compileTemplate(template: string, data: {}) {
  const source = templateToString(template);
  const md: HandlebarsTemplateDelegate = Handlebars.compile(source);
  return new Handlebars.SafeString(md(data));
}

export function safeString(data: string) {
  return new Handlebars.SafeString(data);
}

export function getAnchorRef(ref: string) {
    return ref.replace(/_|\/|\.| /g, '-').replace(/"/g, '').replace(/ /g, '-').toLowerCase();
  }

function templateToString(template: string) {
  return fs.readFileSync(path.join(__dirname, `/${template}`)).toString();
}
