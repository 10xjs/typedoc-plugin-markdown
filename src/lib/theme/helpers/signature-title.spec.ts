import { Fixture, getFixture, getReflectionByName } from '../../../test/test.utils';
import { signatureTitle } from './signature-title';

describe(`signatureTitle helper`, () => {
  test(`should compile`, () => {
    const data = getReflectionByName(getFixture(Fixture.Function), 'functionWithParameters');
    const result = signatureTitle.call(data.signatures[0]);
    expect(result).toMatchSnapshot();
  });
});
