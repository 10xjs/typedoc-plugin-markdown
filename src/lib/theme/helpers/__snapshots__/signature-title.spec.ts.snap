// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signatureTitle helper should compile 1`] = `
"**functionWithParameters**(\`paramZ\`: *\`[object Object]\`*, \`paramG\`: *\`[object Object]\`*, \`paramA\`: *\`[object Object]\`*): *\`[object Object]\`*
"
`;
